File: code/py/real_data/us_stocks/implement_script/changeAUC.py
import sys, random
import pandas as pd
import pickle as pkl
import os
sys.path.insert(0, "./code/py")
if os.getenv("SLURM_SUBMIT_HOST") == "pronto.las.iastate.edu":
    root_dir = "/work/LAS/zhanruic-lab/rohitk/git_repos_data/changeAUC/"
    sys.path.insert(0, "/work/LAS/zhanruic-lab/rohitk/git_repos_data/changeAUC/output")
    sys.path.insert(0, "/work/LAS/zhanruic-lab/rohitk/git_repos_data/changeAUC/data")
elif os.getenv("SLURM_SUBMIT_HOST") == "hpc2021":
    root_dir = "/lustre1/u/rohitisu/git_repos_data/changeAUC/"
    sys.path.insert(0, "/lustre1/u/rohitisu/git_repos_data/changeAUC/output")
    sys.path.insert(0, "/lustre1/u/rohitisu/git_repos_data/changeAUC/data")
else:
    root_dir = ""
    sys.path.insert(0, "./output")
    sys.path.insert(0, "./data")
from get_change_point.get_multiple_change_point_v1 import get_multiple_change_point, get_sbs_cp


dat = pd.read_csv("data/us_stocks/stable_stocks.csv",
                  dtype=float, index_col=0, parse_dates=True)

random.seed(100)
stock_multiple = get_multiple_change_point(dat.values, classifier="FNN", no_of_perm=199, min_length=500)

out_dir = 'output/real_data/us_stocks/'
with open(out_dir + 'fnn_sbs_us_stocks.pkl', "wb") as fp:
    pkl.dump(stock_multiple, fp)

File: code/r/real_data/us_stocks/implement_script/changeAUC.R
source("code/r/get_change_point/get_multiple_change_point_v1.R")

if(Sys.getenv("SLURM_SUBMIT_HOST") == "pronto.las.iastate.edu"){
  out.dir <- "/work/LAS/zhanruic-lab/rohitk/git_repos_data/changeAUC/"
} else if(Sys.getenv("SLURM_SUBMIT_HOST") == "hpc2021"){
  out.dir <- "/lustre1/u/rohitisu/git_repos_data/changeAUC/"
} else{
  out.dir <- ""
}


dat <- read.csv(paste(out.dir,
                      "data/us_stocks/stable_stocks.csv", sep = ""), 
                row.names = 1, header = TRUE)
n <- nrow(dat)
set.seed(100)
output_rf <- get_multiple_change_point(as.matrix(dat), left = 1, right = nrow(dat),
                                    classifier = "RF", 
                                    split_trim = 0.15,
                                    auc_trim = 0.05,
                                    no_of_perm = 199,
                                    min_length = 500, decay = sqrt(2),
                                    return_output = "all")

saveRDS(output_rf,
        paste(out.dir, "output/real_data/us_stocks/rf_sbs_us_stocks.RData", 
              sep = ""))

set.seed(100)
output_reg_logis <- get_multiple_change_point(as.matrix(dat), left = 1, 
                                              right = nrow(dat),
                                       classifier = "REG_LOGIS", 
                                       split_trim = 0.15,
                                       auc_trim = 0.05,
                                       no_of_perm = 199,
                                       min_length = 500, decay = sqrt(2),
                                       return_output = "all")

saveRDS(output_reg_logis,
        paste(out.dir, "output/real_data/us_stocks/rf_sbs_us_stocks.RData", 
              sep = ""))
